config/src/main/java/org/springframework/security/config/annotation/web/reactive/ServerHttpSecurityConfiguration.java

```diff
@@ -43,7 +43,7 @@
  * @since 5.0
  */
 @Configuration
-class ServerHttpSecurityConfiguration implements WebFluxConfigurer {
+class ServerHttpSecurityConfiguration {
```

Comment:
Can you explain why we need to expose this via a Bean vs the Configuration? Does setting proxy proxyBeanMethods mean the `ServerHttpSecurityConfiguration` is no longer registered as a Bean?

Reply:
`ServerHttpSecurityConfiguration` will still be registered as a Bean. 
The reason for this change is to avoid calling `authenticationPrincipalArgumentResolver()` from the `configureArgumentResolvers` override. 
I noticed spring boot had a similar situation when disabling proxying and this was their approach 
https://github.com/spring-projects/spring-boot/blob/dfb2e8a278e7211c658104ae45a19b68aec57a21/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfiguration.java#L136

Reply:
Ok thanks for the clarification. I think that this makes sense.